### PR TITLE
Add support for deleting threepids

### PR DESCRIPTION
**Pull request by dbkr**
Requires https://github.com/vector-im/riot-web/pull/2784
Requires https://github.com/matrix-org/matrix-js-sdk/pull/312

### CODE REVIEW

File: src/components/structures/UserSettings.js [hunk 1]
```diff
@@ -248,6 +248,31 @@ module.exports = React.createClass({
...
+            onFinished: (submit) => {
+                if (submit) {
+                    this.setState({
+                        phase: "UserSettings.LOADING",
+                    });
+                    MatrixClientPeg.get().deleteThreePid(threepid.medium, threepid.address).then(() => {
+                        return this._refreshFromServer();
+                    }).catch((e) => {
+                        const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
+                        Modal.createDialog(ErrorDialog, {
```
Comment (ara4n) [inline src/components/structures/UserSettings.js]: will creating a modal from within an existing modal work? I vaguely remember having to put the second create into a setTimeout(0) to let the other one first disappear...?

File: src/components/structures/UserSettings.js [hunk 2]
```diff
@@ -507,7 +531,7 @@ module.exports = React.createClass({
             this.state.avatarUrl ? MatrixClientPeg.get().mxcUrlToHttp(this.state.avatarUrl) : null
         );
 
-        var threepidsSection = this.state.threepids.map(function(val, pidIndex) {
+        var threepidsSection = this.state.threepids.map((val, pidIndex) => {
             var id = "email-" + val.address;
```
Comment (ara4n) [inline src/components/structures/UserSettings.js]: `3pid-` ?

**Comment by ara4n** [comment]
lgtm mod comments

File: src/components/structures/UserSettings.js (hunk 1 above)
Comment (dbkr) [inline src/components/structures/UserSettings.js]: Empirically it seems to be fine (although I did get the variable name wrong)

**Pull request merged**